### PR TITLE
Release instructions 1.12

### PR DESCRIPTION
**Pull request by nitinjainsj**
Update release instructions. These are what worked for 1.12 release and setting up master for 2.0

### CODE REVIEW

File: doc/major_minor_release_instructions.md [hunk 1]
```diff
@@ -43,75 +43,95 @@ make regenerate-test-deploy-manifests
...
 
 * Change the package names to be suffixed with the version, e.g. in
-  `pfs.proto` change `package pfs`; to `package pfs_1_11;`. Also update the
+  `pfs.proto` change `package pfs`; to `package pfs_1_12;`. Also update the
   `go_package` field.
 * Update the imports to reference the copied protos. e.g. in `pps.proto`,
   change `import client/pfs/pfs.proto` to
-  `import client/admin/v1_11/pfs/pfs.proto`, and any uses of `pfs.` to
-  `pfs_1_11.`.
+  `import client/admin/v1_12/pfs/pfs.proto`, and any uses of `pfs.` to
```
Comment (msteffen) [inline doc/major_minor_release_instructions.md:71]: Nit: stray period after `pfs`

File: doc/major_minor_release_instructions.md [hunk 2]
```diff
@@ -43,75 +43,95 @@ make regenerate-test-deploy-manifests
...
+  `pfs_1_12.`.
 
 In `src/client/admin/admin.proto`:
 
 * Import the copied PFS and PPS protos
-* Rename `Op1_11` (which should already exist) to `Op1_12`.
-* Add a new `Op1_11`, which is basically a copy of `Op1_12` referencing the
+* Rename `Op1_12` (which should already exist) to `Op2_0`.
+* Add a new `Op1_12`, which is basically a copy of `Op2_0` referencing the
```
Comment (msteffen) [inline doc/major_minor_release_instructions.md:78]: Nit: should this be "add a new `Op2_0`, which is basically a copy of `Op1_12`"?

File: doc/major_minor_release_instructions.md [hunk 3]
```diff
@@ -43,75 +43,95 @@ make regenerate-test-deploy-manifests
...
+
 Finally, run `make proto` to rebuild the protos.
 
 #### Wire in the proto changes
 
 Copy over the converters:
 
 ```shell
-cp src/server/admin/server/convert1_10.go src/server/admin/server/convert1_11.go
+cp src/server/admin/server/convert1_11.go src/server/admin/server/convert1_12.go
```
Comment (msteffen) [inline doc/major_minor_release_instructions.md:111]: Nit: is this a version behind? I.e. should it be `cp src/server/admin/server/convert1_12.go src/server/admin/server/convert2_0.go`?

File: doc/major_minor_release_instructions.md [hunk 4]
```diff
@@ -43,75 +43,95 @@ make regenerate-test-deploy-manifests
...
 ```
 
 Then update `convert1_11.go`:
 
+* Update imports for pfs, pps, auth, enterprise to point to 1.12 version. Eg. `import client/pfs` to `import client/admin/v1_12/pfs`
```
Comment (msteffen) [inline doc/major_minor_release_instructions.md:116]: Likewise, should this be `Update imports...to point to the 2.0 version`?

File: doc/release_instructions.md [hunk 5]
```diff
@@ -126,16 +118,30 @@ git push
...
-Then update the
-[release's notes](https://github.com/pachyderm/pachyderm/releases), and post
-the update on the #users channel.
+* To release a major, minor, or patch version, run
+```shell
+make point-release
+```
+* To release an alpha/beta/RC version, specify the additional text to appear in the release version and run
+```shell
+ make VERSION_ADDITIONAL=-alpha1 release-candidate
```
Comment (msteffen) [inline doc/release_instructions.md:127]: Nit: the lines in this code block (`make`, `OR`, etc) seem like they have leading spaces?

File: doc/major_minor_release_instructions.md [hunk 6]
```diff
@@ -43,75 +43,95 @@ make regenerate-test-deploy-manifests
...
   just the above replacements are sufficient)
 
 Update the admin client (`src/client/admin.go`) and admin server
 (`src/server/admin/server/api_server.go`). Mostly this just involves looking
-for instances of `1_10` or `1_11` in those files and updating the code.
+for instances of `1_11` or `1_12` in those files and updating the code and 
+add `applyOp2_0` function which should be a copy of `applyOp1_12`
 
-Update `convert1_7.go`, around line 309, to include `convert1_11Objects`.
+Update `convert1_7.go`, around line 309, to include `convert1_12Objects`.
```
Comment (msteffen) [inline doc/major_minor_release_instructions.md:134]: "...to include `convert2_0Objecs`"?

**Review by msteffen — APPROVED** [review]
A few nits but mostly LGTM

File: doc/major_minor_release_instructions.md (hunk 1 above)
Comment (nitinjainsj) [reply doc/major_minor_release_instructions.md:71]: I think this is intentional.

File: doc/major_minor_release_instructions.md (hunk 2 above)
Comment (nitinjainsj) [reply doc/major_minor_release_instructions.md:78]: It was confusing to me as well. The "Add" step is after rename. So Op1_12 is now Op2_0

File: doc/major_minor_release_instructions.md (hunk 3 above)
Comment (nitinjainsj) [reply doc/major_minor_release_instructions.md:111]: No. This is for 1.12 release.

File: doc/major_minor_release_instructions.md (hunk 4 above)
Comment (nitinjainsj) [reply doc/major_minor_release_instructions.md:116]: 2.0 version does not exist. 2.0 is the head version `client/pfs` `client/pps`, etc.

File: doc/major_minor_release_instructions.md (hunk 6 above)
Comment (nitinjainsj) [reply doc/major_minor_release_instructions.md:134]: No. This conversion is for 1_12. 2.0 conversion does not exist yet. We are getting master ready for eventually writing the 2.0 conversion when we do 2.x branch.

**Review by nitinjainsj — COMMENTED** [review]
The set of instructions are confusing due to the new-1 release, new release, future release references.

FWIW, the instructions are still not very clear, but it's a step towards adding more inf.

**Pull request merged**